pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -81,21 +80,29 @@ def restore_weights(self, model: LightningModule):
         if self.trainer.on_gpu:
             torch.cuda.empty_cache()
 
-    def restore(self, checkpoint_path: str, on_gpu: bool):
+    def restore(self, checkpoint_path: str, on_gpu: bool) -> bool:
         """
-        Restore training state from checkpoint.
+        Try to restore training state from checkpoint.
         Also restores all training state like:
         - epoch
         - callbacks
         - schedulers
         - optimizer
+
+        Returns:
+           `True` if restored successfully else `False`
         """
 
-        # if on_gpu:
-        #     checkpoint = torch.load(checkpoint_path)
-        # else:
-        # load on CPU first
-        checkpoint = pl_load(checkpoint_path, map_location=lambda storage, loc: storage)
+        # Try to load checkpoint from `checkpoint_path`. If failed, do not restore checkpoint.
+        try:
```

Comment:
I wonder if it would be better to be explicit with the check rather than catching all exceptions (there could be nasty side effects from catching all here). Maybe replacing this with a `os.path.exists(...)` check rather than a try except to enable restoring?

Reply:
Great idea!
I agree with you, so will fix it quickly.

Reply:
Fixed it with the method similar to `restore_hpc_weights_if_needed` in `CheckpointConnector`.